stl/inc/atomic

```diff
@@ -355,13 +370,27 @@ struct _Atomic_storage {
         const auto _Storage_ptr  = _STD addressof(_Storage);
         const auto _Expected_ptr = _STD addressof(_Expected);
         bool _Result;
+#if _CMPXCHG_MASK_OUT_PADDING_BITS
+        __builtin_zero_non_value_bits(_Expected_ptr);
+#endif // _CMPXCHG_MASK_OUT_PADDING_BITS
         _Lock();
-        if (_CSTD memcmp(_Storage_ptr, _Expected_ptr, sizeof(_Ty)) == 0) {
+#if _CMPXCHG_MASK_OUT_PADDING_BITS
+        if constexpr (_Cmpxchg_has_padding_bits_v<_Ty>) {
+            _Ty _Local;

```

Comment:
`_Ty` might not be default constructible so we can't do this. Ditto below for the `_Mask`s. You can either use a union or borrow `_Storage_for` from `<execution>`.

(Sorry I missed this before)

Reply:
I fixed.